tests/unittests/sources/test_azure.py

```diff
@@ -1055,7 +1055,6 @@ def setUp(self):
                 mock.MagicMock(),
             )
         )
-        super(TestAzureDataSource, self).setUp()
```

Comment:
Why remove the call to the parent class setUp?

Reply:
Because line https://github.com/canonical/cloud-init/blob/e168b4a1383b6eae9c1dc81411d7684fcbbf7df9/tests/unittests/sources/test_azure.py#L1014 already calls this, and doubly calling `CiTestCase.setUp` makes losing the old logging handlers in https://github.com/canonical/cloud-init/blob/e168b4a1383b6eae9c1dc81411d7684fcbbf7df9/tests/unittests/helpers.py#L153, which are not restored in `CiTestCase.tearDown`. In particular, the LogExporter handler gets removed provoking other subsequent test errors:

```
FAILED tests/unittests/test_cli.py::TestCLI::test_status_wrapper_init_local_writes_fresh_status_info - RuntimeError: generator raised StopIteration
FAILED tests/unittests/test_cli.py::TestCLI::test_status_wrapper_init_local_honor_cloud_dir - RuntimeError: generator raised StopIteration
```

Those Exceptions are raised from https://github.com/canonical/cloud-init/blob/e168b4a1383b6eae9c1dc81411d7684fcbbf7df9/cloudinit/cmd/main.py#L770-L772

Reply:
I missed the duplicate - nice catch!